### ISSUE TITLE
More details when resource fails to load

### ISSUE THREAD

**Issue by VladimirMarkelov** [17cupsofcoffee/tetra#127]
Is it possible to add filename to the error text when Tetra cannot load a resource from file?

At this moment, if I make a typo in path or filename in a command like `Texture::new(ctx, "./res/image.png")?` An error printed in terminal is:
```
Error: Image(IoError(Os { code: 2, kind: NotFound, message: "The system cannot find the file specified." }))
```

I was lucky I added only one resource when it failed and I just checked its path. But in general case it may require reading and checking all the used paths/filenames.

**Comment by 17cupsofcoffee** [comment]
Huh, I'm surprised that the `io::Error` message returned by Rust doesn't include this!

In an ideal world, I imagine Tetra's error types would be a bit more helpful than this - something like an `TetraError::AssetFailedToLoad`, which would contain the context of *which* asset failed, as well as the `std::io::Error` as the cause. In general, I feel like Tetra's error handling is a bit weak, and I wish I'd spent more time on it before 0.1!

Fixing these kinds of things is one of the main areas I want to focus on in 0.3 - I'll see if there's anything I can do in the meantime without making a breaking change, though.

**Comment by 17cupsofcoffee** [comment]
This is now implemented on master as `TetraError::FailedToLoadAsset` :tada: (name subject to change!)

The slight downside of this change is that it means there's an extra allocation when a texture fails to load. But I figure that in practice, you'd never expect a texture to be missing in your game when it's released - the slight overhead is worth it for better diagnostics.

I also noticed while implementing the change that a missing texture returns a `TetraError::Image` instead of a `TetraError::Io`, which isn't what I'd expect - the behaviour should be consistent for all asset types now.
